### PR TITLE
Uc 229 fixing Special:UserSignup on mobile

### PR DESCRIPTION
**Pull request by lizlux**
Fixing JS errors on mobile signup page that started after some recent changes we made to the oasis version of the form. 

Fixes: 
- Captcha check now disregards mobile b/c there's no captcha on mobile at the moment
- Marketing opt-in and tracking of the country code are now set up on mobile. The JS was expecting those inputs to be there and they weren't. 
- Checkboxes weren't styled properly on mobile. I guess we didn't have any before this!

https://wikia-inc.atlassian.net/browse/UC-229

### CODE REVIEW

File: extensions/wikia/UserLogin/templates/UserSignupSpecial_WikiaMobileIndex.php [hunk 1]
```diff
@@ -1,92 +1,110 @@
...
 				'isRequired' => true,
 				'class' => 'birthday-group',
-				'isInvalid' => (!empty($errParam) && $errParam === 'birthyear') || (!empty($errParam) && $errParam === 'birthmonth') || (!empty($errParam) && $errParam === 'birthday'),
-				'errorMsg' => (!empty($msg) ? $msg : ''),
-				'params' => array('birthyear' => $birthyear, 'birthmonth' => $birthmonth, 'birthday' => $birthday, 'isEn' => $isEn),
-			),
-			array(
+				'isInvalid' => ( !empty( $errParam ) && $errParam === 'birthyear' ) ||
+					( !empty( $errParam ) && $errParam === 'birthmonth' ) ||
+					( !empty( $errParam ) && $errParam === 'birthday' ),
```
Comment (armonr) [inline extensions/wikia/UserLogin/templates/UserSignupSpecial_WikiaMobileIndex.php]: While you're at it, would you mind cleaning the logic to the following please?
```php
'isInvalid' => ( !empty( $errParam ) &&
				( $errParam === 'birthyear' || $errParam === 'birthmonth' || $errParam === 'birthday' ) ),
```

File: extensions/wikia/UserLogin/templates/UserSignupSpecial_WikiaMobileIndex.php [hunk 2]
```diff
@@ -1,92 +1,110 @@
...
-			array( //fake username field (not in use)
+			],
+			[ //fake username field ( not in use )
 				'type' => 'hidden',
 				'name' => 'username',
 				'value' => '',
-				'label' => wfMessage('yourname')->text(),
-			),
-			array( //actual username field
+				'label' => wfMessage( 'yourname' )->text(),
```
Comment (jsutterfield) [inline extensions/wikia/UserLogin/templates/UserSignupSpecial_WikiaMobileIndex.php]: I believe this should be escaped. Also lines 22, 31, 40 and 46,

File: extensions/wikia/UserLogin/templates/UserSignupSpecial_WikiaMobileIndex.php [hunk 3]
```diff
@@ -1,92 +1,110 @@
...
+				'isInvalid' => ( !empty( $errParam ) && $errParam === 'birthyear' ) ||
+					( !empty( $errParam ) && $errParam === 'birthmonth' ) ||
+					( !empty( $errParam ) && $errParam === 'birthday' ),
+				'errorMsg' => ( !empty( $msg ) ? $msg : '' ),
+				'params' => [
+					'birthyear' => $birthyear,
+					'birthmonth' => $birthmonth,
+					'birthday' => $birthday,
+					'isEn' => $isEn
+				],
```
Comment (jsutterfield) [inline extensions/wikia/UserLogin/templates/UserSignupSpecial_WikiaMobileIndex.php]: Having these two ```],``` stacked on top of eachother looks kind of weird. How about either indenting the top one, or putting it right after ```'isEn' => $isEn```?

File: extensions/wikia/UserLogin/templates/UserSignupSpecial_WikiaMobileIndex.php [hunk 4]
```diff
@@ -1,92 +1,110 @@
...
+			],
```
Comment (jsutterfield) [inline extensions/wikia/UserLogin/templates/UserSignupSpecial_WikiaMobileIndex.php]: Having these two ```],``` stacked on top of eachother looks kind of weird. How about either indenting the top one, or putting it right after ```'isEn' => $isEn```?

**Comment by jsutterfield** [comment]
Also, don't forget to link to the jira ticket in the PR description.

File: extensions/wikia/UserLogin/templates/UserSignupSpecial_WikiaMobileIndex.php (hunk 1 above)
Comment (lizlux) [inline extensions/wikia/UserLogin/templates/UserSignupSpecial_WikiaMobileIndex.php]: Ha, yes will do thanks.

File: extensions/wikia/UserLogin/templates/UserSignupSpecial_WikiaMobileIndex.php (hunk 4 above)
Comment (lizlux) [inline extensions/wikia/UserLogin/templates/UserSignupSpecial_WikiaMobileIndex.php]: Not sure what you mean. The indentation looks okay to me.

Comment (jsutterfield) [inline extensions/wikia/UserLogin/templates/UserSignupSpecial_WikiaMobileIndex.php]: I think of brackets here like I think of braces for statements. They give visual cues for what the scope is. The above reminds me of code like this.
```php
function foo() {
    if ( someCondition) {
    // do stuff
}
}
```
Having the 'params' closing bracket be indented at the same level as the closing bracket for the array which 'params' is a part of looks awkward to me. It's up to you, but that's my perspective as someone reading the code.

**Comment by jsutterfield** [comment]
LGTM!

**Comment by armonr** [comment]
:+1:

**Pull request merged**